### PR TITLE
Adding jenkins fedora slave

### PR DESCRIPTION
**Pull request by surajnarwade**
This is fedora jenkins slave dockerfile which can be used when docker enviroment is used as slave with jenkins along with docker plugin.

### CODE REVIEW

**Comment by scollier** [comment]
Hi @surajnarwade How do I test that this works?  Can you please include some simple instructions?  Thanks.

**Comment by surajnarwade** [comment]
Hi @scollier , thanks for reply, here is instructions to be followed:
1. install docker plugin in jenkins
2.  add this line -H fd:// -H tcp://0.0.0.0:2376 in addition to ExecStart, so it should look like this,
ExecStart=/usr/bin/docker daemon -H fd:// -H tcp://0.0.0.0:2376
3. sudo systemctl  daemon-reload
4. sudo systemctl restart docker
5.goto jenkins> manage jenkins > configure system > Add new cloud
6.select docker cloud, fill required details docker URL(<ip>:2376), label, remote filling system to /home/jenkins , docker image name(this docker image) and start to configure job.
7. put label of docker cloud in restrict where this project can be run. and build the job.

Thanks.

**Comment by scollier** [comment]
Hi @surajnarwade please add those instructions to the README, so that when people clone the repo, they know what to do and don't have to refer to this PR.

Thanks.

File: jenkins-slave/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM fedora:latest
+MAINTAINER http://fedoraproject.org/wiki/Cloud
+
+RUN dnf -y update && dnf clean all
+RUN dnf -y install java-1.8.0-openjdk openssh-server
```
Comment (scollier) [inline jenkins-slave/Dockerfile]: Please add a clean all to this.  you have to clean every layer that you run dnf install on.

File: jenkins-slave/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+RUN ssh-keygen -A
+RUN sed -i 's|session    required     pam_loginuid.so|session    optional     pam_loginuid.so|g' /etc/pam.d/sshd
+
+RUN mkdir -p /var/run/sshd
+RUN useradd jenkins
+RUN echo "jenkins:jenkins" | chpasswd
```
Comment (baude) [inline jenkins-slave/Dockerfile]: Should this be done during the docker build or through an install script before the user runs it?

File: jenkins-slave/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+
+
+EXPOSE 22
+
+CMD ["/usr/sbin/sshd", "-D"]
+
```
Comment (baude) [inline jenkins-slave/Dockerfile]: Should we add INSTALL and RUN labels to this for atomic use?

File: jenkins-slave/Dockerfile (hunk 1 above)
Comment (surajnarwade) [inline jenkins-slave/Dockerfile]: Hi @scollier, I have updated README file as well added clean all in every layer.
Thanks

File: jenkins-slave/Dockerfile (hunk 3 above)
Comment (surajnarwade) [inline jenkins-slave/Dockerfile]: @baude , I have written this for fedora cloud image, as we use it as for jenkins slave, atomic are not preferred for jenkins slave.
Thanks

File: jenkins-slave/Dockerfile (hunk 2 above)
Comment (surajnarwade) [inline jenkins-slave/Dockerfile]: @baude During docker build

**Comment by surajnarwade** [comment]
@scollier @baude  I have updated the patchset.
Thanks

**Pull request merged**